### ISSUE TITLE
Display string technical instruction and edit fingerings

### ISSUE THREAD

**Issue by jmorag** [opensheetmusicdisplay/opensheetmusicdisplay#711]
I'm working on an database of violin fingerings for common concerti and excerpts. OSMD works well to display fingerings, but the `<string>` attribute inside `<technical>` doesn't render. 

Adding 
```typescript
    const nodeString: IXmlElement = xmlNode.element("string");
    if (nodeString !== undefined) {
      const currentTechnicalInstruction: TechnicalInstruction = new TechnicalInstruction();
      currentTechnicalInstruction.type = TechnicalInstructionType.String;
      currentTechnicalInstruction.value = nodeString.value;
      switch (nodeString.value) {
        case "1":
          currentTechnicalInstruction.value = "I";
          break;
        case "2":
          currentTechnicalInstruction.value = "II";
          break;
        case "3":
          currentTechnicalInstruction.value = "III";
          break;
        case "4":
          currentTechnicalInstruction.value = "IV";
          break;
      }
      currentTechnicalInstruction.placement = PlacementEnum.Left;
      currentVoiceEntry.TechnicalInstructions.push(currentTechnicalInstruction);
    }
```
to `addTechnicalArticulations` throws a vexflow error:
```
index.js:675 Error loading sheet: TypeError: Cannot read property 'line' of undefined
StackTrace: 
TypeError: Cannot read property 'line' of undefined
    at Function.value (vexflow-debug.js:1)
    at vexflow-debug.js:1
    at Array.forEach (<anonymous>)
    at t.value (vexflow-debug.js:1)
    at e.value (vexflow-debug.js:1)
    at e.value (vexflow-debug.js:1)
    at vexflow-debug.js:1
    at Array.map (<anonymous>)
    at t.value (vexflow-debug.js:1)
    at VexFlowMusicSheetCalculator.calculateMeasureXLayout (VexFlowMusicSheetCalculator.ts:133)
```
Is there any plan to support rendering the `<string>` tag or is this something that vexflow simply can't support?

My second question (apologies if this should be two separate issues) is, is it possible to interactively edit fingerings for notes under the cursor? Setting `this.osmd.cursor.VoicesUnderCursor()[0].TechnicalInstructions[0] = "4"`, for example, doesn't show any change in the displayed output, even after calling `osmd.render()`.

**Comment by CameronRKing** [comment]
I've been coming at the same problem from a different angle, trying to generate sight reading for the guitar. To generate the string number graphically, I went into `VexFlowMeasure@createFingerings` and added a simple switch statement to differentiate between `FretHandFingers` and `StringNumbers`. Then I hit the same error you did.

After some thought, I realized that the issue is the call to `vexFlowVoiceEntry.vfStaveNote.addModifier((i as any), (fretFinger as any))`. The example file I was using for testing had both `<fingering>` and  `<string>`  technical elements. When `createFingerings` loops over them, `i` gets incremented. The use of `i` in `vfStaveNote.addModifier` indicates which note the modifier applies to.  However, in my example, there's only one note. The second modifier gets attached to a note that doesn't exist. Hence the error. I can dodge it by hard-coding `0` into the call, but that won't work for chords.

I've only been tinkering with MusicXML for about a day, so I'm not yet sure how to implement a robust solution that handles chords with selective fingering/string designations. I'll keep playing with it.

**Comment by sschmidTU** [comment]
@CameronRKing ah, interesting, thanks for the details, makes it much easier to investigate. I'll take a look.

> My second question (apologies if this should be two separate issues) is, is it possible to interactively edit fingerings for notes under the cursor? Setting this.osmd.cursor.VoicesUnderCursor()[0].TechnicalInstructions[0] = "4", for example, doesn't show any change in the displayed output, even after calling osmd.render().

That's a bit difficult, because the fingerings are added before rendering, and we don't have a system to delete and re-render individual vexflow elements without re-rendering the whole piece. This is a general theme that would probably be a great next step for OSMD, also performance-wise.
So, for now, you'd probably have to set osmd.graphic to null and call render(), rendering the whole sheet again.

**Comment by sschmidTU** [comment]
@CameronRKing the fingering bug [you mentioned](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/711#issuecomment-611635185) is now fixed (in ee80e911c85e33bb1c96daf14be9af4420e59c22). We did count all technical instructions as fingerings and overlooked that there are also `<string>` and `<fret>` instructions. It should now work correctly.
We're also preparing a new OSMD 0.7.4 release right now.

**Comment by sschmidTU** [comment]
You can already change the fingering without re-rendering by searching the SVG for `<text>` nodes, but I just made it easier by giving fingerings a reference to their SVG node.
```js
const fingeringSVG = osmd.graphic.MeasureList[0][0].staffEntries[0].FingeringEntries[0].SVGNode;
fingeringSVG.children[0].textContent = 4; // change fingering e.g. from 5 to 4

// remove fingering:
osmd.drawer.backend.getSvgElement().removeChild(fingeringSVG);
// alternatively, make the fingering transparent, so you can show it again (un-remove) later
```

I also gave all other labels a reference to their SVGNode, e.g. lyrics, measure numbers, page labels, dynamics, chord symbols, etc., so it is easier to manipulate all of them in the SVG without re-rendering now.
